ui/src/utils/AudioRecorder.js

```diff
@@ -76,9 +75,7 @@ export class AudioRecorder {
                     resolve(audio);
                 };
             });
-        } else {
-            alert('Error: Could not record successfully.');
-            return null;
         }
+        throw new Error('Could not record successfully.');
```

Comment:
`dispatchErrorToast` ?

Reply:
I don't think a library should be dispatching events. It should throw an error and be caught by the components which then dispatch events.